skiko/src/commonMain/kotlin/org/jetbrains/skia/FontMgr.kt

```diff
@@ -10,6 +11,13 @@ open class FontMgr : RefCnt {
         }
 
         val default = FontMgr(_nDefault(), false)
+
+        fun defaultWithFallbackFontProvider(fallback: TypefaceFontProvider): FontMgr {
```

Comment:
The functionality of adding fallback seems doesn't relate to the default FontMgr, and the API looks unnecessary restricted from an external user perspective (allowing fallbacks only for the default manager).

I would make a separate extended class instead:
```
class FontMgrWithFallback: FontMgr() {
    constructor(fallback: TypefaceFontProvider): this(_nDefaultWithFallbackFontProvider(fallback._ptr))
}
```

This also mimics the C++ API, which is a good thing for keeping Skiko API as close as possible to the original.

User can use the API this way then:
```
val fallbackProvider = TypefaceFontProvider().apply {
    registerTypeface(notoEmojiTypeface)
}

fontCollection.setDefaultFontManager(FontMgrWithFallback(FontMgr.default, fallbackProvider))
```

Reply:
The usage is a bit different though:

the default `TypefaceFontProvider` won't provide the fallbacks.
But `TypefaceFontProvider.createAsFallbackProvider()` will provide them.

Reply:
- add a new `TypefaceFontProviderWithFallback` it extends the existing `TypefaceFontProvider`
- `TypefaceFontProvider` is now an open class
- added a new`class FontMgrWithFallback(fallbackProvider: TypefaceFontProviderWithFallback)`, which relies on skia's FontMgr.default under the hood + our new fallback provider